File: mobile/src/main/java/treehou/se/habit/ui/settings/subsettings/wiget/WidgetSettingsContract.java
package treehou.se.habit.ui.settings.subsettings.wiget;

import treehou.se.habit.mvp.BasePresenter;
import treehou.se.habit.mvp.BaseView;

public interface WidgetSettingsContract {

    interface View extends BaseView<Presenter> {
        void setWidgetBackground(int backgroundType);
        void setWidgetTextSize(int size);
        void setWidgetImageSize(int size);
        void setCompressedWidgetButton(boolean isChecked);
        void setCompressedWidgetSlider(boolean isChecked);
    }

    interface Presenter extends BasePresenter {
        void setWidgetBackground(int backgroundType);
        void setWidgetTextSize(int size);
        void setWidgetImageSize(int size);
        void setCompressedWidgetButton(boolean isChecked);
        void setCompressedWidgetSlider(boolean isChecked);
    }
}

File: mobile/src/main/java/treehou/se/habit/ui/settings/subsettings/wiget/WidgetSettingsContract.kt
package treehou.se.habit.ui.settings.subsettings.wiget

import treehou.se.habit.mvp.BasePresenter
import treehou.se.habit.mvp.BaseView

interface WidgetSettingsContract {

    interface View : BaseView<Presenter> {
        fun setWidgetBackground(backgroundType: Int)
        fun setWidgetTextSize(size: Int)
        fun setWidgetImageSize(size: Int)
        fun setCompressedWidgetButton(isChecked: Boolean)
        fun setCompressedWidgetSlider(isChecked: Boolean)
    }

    interface Presenter : BasePresenter {
        fun setWidgetBackground(backgroundType: Int)
        fun setWidgetTextSize(size: Int)
        fun setWidgetImageSize(size: Int)
        fun setCompressedWidgetButton(isChecked: Boolean)
        fun setCompressedWidgetSlider(isChecked: Boolean)
    }
}
